system/keyboard.c

```diff
@@ -216,6 +223,133 @@ static const char usb_hid_keymap[] = {
 
 keyboard_types_t keyboard_types = KT_NONE;
 
+//------------------------------------------------------------------------------
+// Private Functions
+//------------------------------------------------------------------------------
+
+// This function is called when an ESC O prefix has been detected.
+static char get_vt220_sequence1(void)
+{
+    switch (tty_get_char(TTY_SEQ_WAIT_TIME)) {
+      case 'P':
+        return '1';  // VT100/VT220 PF1 (F1 in terminal emulators)
+      case 'Q':
+        return '2';  // VT100/VT220 PF2 (F2 in terminal emulators)
+      case 'R':
+        return '3';  // VT100/VT220 PF3 (F3 in terminal emulators)
+      case 'S':
+        return '4';  // VT100/VT220 PF4 (F4 in terminal emulators)
+      case 'T':
+        return '5';  // F5 in some terminal emulators
+      case 'U':
+        return '6';  // F6 in some terminal emulators
+      case 'V':
+        return '7';  // F7 in some terminal emulators
+      case 'W':
+        return '8';  // F8 in some terminal emulators
+      case 'X':
+        return '9';  // F9 in some terminal emulators
+      case 'Y':
+        return '0';  // F10 in some terminal emulators
+      default:
+        return '\0'; // unrecognised key
+        break;
+    }
+}
+
+// This function is called when an ESC [ prefix has been detected.
+static char get_vt220_sequence2(void)
+{
+    char ch1;
+    char ch2;
+
+    ch1 = tty_get_char(TTY_SEQ_WAIT_TIME);
+    switch (ch1) {
+      case 'A':
+        return 'u';  // VT100/VT220 cursor up
+      case 'B':
+        return 'd';  // VT100/VT220 cursor down
+      case 'C':
+        return 'r';  // VT100/VT220 cursor right
+      case 'D':
+        return 'l';  // VT100/VT220 cursor left
+      default:
+        break;
+    }
+    if ((ch1 < '1') || (ch1 > '6')) {
+        return '\0';  // unrecognised sequence - give up
+    }
+    ch2 = tty_get_char(TTY_SEQ_WAIT_TIME);
+    if (ch2 == '~') {
+        return '\0';  // VT100/VT220 editing key - ignore it
+    }
+    switch (ch1) {
+      case '1':
+        switch (ch2) {
+          case '1':
+            ch1 = '1'; break;  // F1 in terminal emulators
+          case '2':
+            ch1 = '2'; break;  // F2 in terminal emulators
+          case '3':
+            ch1 = '3'; break;  // F3 in terminal emulators
+          case '4':
+            ch1 = '4'; break;  // F4 in terminal emulators
+          case '5':
+            ch1 = '5'; break;  // F5 in terminal emulators
+          case '7':
```

Comment:
1-5 map directly, and 7-9 map to 6-8... that seems cursed, but so be it, I guess :)